[AR-526] Moving mvebu-dev to lk5.9

heisath opened the pull request:
Hi,

this PR will move the mvebu-dev kernel to 5.9 for testing and fixing prior to moving mvebu-current to 5.9 later. 

There are still some problems with 5.9, so this is only a draft. 

Building fails with:
```
arch/arm/vfp/vfphw.o: in function `vfp_support_entry':
(.text+0xa): relocation truncated to fit: R_ARM_THM_JUMP19 against symbol `vfp_kmode_exception' defined in .text.unlikely section in arch/arm/vfp/vfpmodule.o
```

Disabling VFP in kernel config solves the errors and compilation goes through, but I think we need/want VFP NEON support for these boards. 

Also there is something broken for the uboot for clearfog-base. But that might just be a config mistake.
-> has been fixed by disabling mainline / dev uboot for clearfog as it is not working anyway.

Please comment and/or help fixing :D

neheb commented:
I knew I saw that error before.

Checking my email shows this: https://git.musl-libc.org/cgit/musl/commit/?id=b1e2aae0aa6edd91413b288b9cc6939f2d80e302

Might be related...

neheb commented:
I would also update 40-pci-add-irq-change-handler-sspl.patch to Russell King's latest version: http://git.arm.linux.org.uk/cgit/linux-arm.git/log/?h=clearfog

heisath commented:
Great finds, I am working on integrating this.

heisath commented:
So mvebu-dev does compile and work on clearfogpro. I will build images and test on clearfog & helios4. Then we can use this as base for bringing mvebu-current up to LK5.9 after the 20.11 release.

neheb commented:
What is this patch? https://github.com/armbian/build/blob/87aa3de5226f26d6e6d02160b1f6aefa0ff9109d/patch/kernel/mvebu-dev/fix_time_drift_remove_global_timer.patch

heisath commented:
This patch is related to a time drift problem we had. See this very old thread. https://forum.armbian.com/topic/6387-fixed-clearfogpro-huge-timedrift-between-hw-and-sysclock/?tab=comments#comment-49029

neheb commented:
Hmm what I got from that is the dynamic frequency scaling patches are messing something up.

I'm comparing the patches here to OpenWrt. We don't carry these.

heisath commented:
I researched this a bit further, and tested these situations:

With DFS patches and global timer disabled (current default):
- dfs working
- no time drift in `timedatectl`

With DFS patches but fixed frequency (governor userspace, freq 1.6GHz) and global timer disabled:
- dfs working, fixed frequency
- no time drift

With DFS patches but fixed frequency (governor userspace, freq 1.6GHz) and global timer enabled (patch removed):
- dfs working, fixed frequency
- slow time drift (visible after 10 minutes)

With DFS patches (governor ondemand, freq 800-1600MHz) and global timer enabled (patch removed):
- dfs working, fixed frequency
- fast time drift (visible after 1 minute)

Without DFS patches (fixed freq 1.6GHz) and global timer disabled:
- dfs not there, fixed frequency
- no time drift

Without DFS patches (fixed freq 1.6GHz) and global timer enabled (patch removed):
- dfs not there, fixed frequency
- no time drift

igorpecovnik reviewed (approved):
I am fine with this PR. Also u-boot can stay behind - i suppose its some work to get it to the mainline and its a question if its worth proceeding that path.

heisath commented:
Mainline uboot would work for helios4 (at least last time I tested) but for clearfog there were serious problems with PCIE (check the Todo issues here: https://github.com/orgs/armbian/projects/3 )   So IMHO not worth it. I will try from time to time, maybe clearfog gets mainlined enough for it to work. Official solidrun image have the same problems btw.
